src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -91,22 +91,25 @@ def send_historical_data_to_seer(alert_rule: AlertRule, project: Project) -> Ale
         config=anomaly_detection_config,
         timeseries=formatted_data,
     )
-    try:
-        make_signed_seer_api_request(
-            connection_pool=seer_anomaly_detection_connection_pool,
-            path=SEER_ANOMALY_DETECTION_STORE_DATA_URL,
-            body=json.dumps(body).encode("utf-8"),
-        )
+    response = make_signed_seer_api_request(
+        connection_pool=seer_anomaly_detection_connection_pool,
+        path=SEER_ANOMALY_DETECTION_STORE_DATA_URL,
+        body=json.dumps(body).encode("utf-8"),
+    )
+    # TODO: this errors if we hit the Exception("Search for optimal window failed.") because response.data is text, not a dict
+    results: StoreDataResponse = json.loads(response.data.decode("utf-8"))
     # See SEER_ANOMALY_DETECTION_TIMEOUT in sentry.conf.server.py
-    except (TimeoutError, MaxRetryError):
-        logger.warning(
-            "Timeout error when hitting Seer store data endpoint",
+    if not results.get("success"):
```

Comment:
We need to update the exception handling in [create_alert_rule](https://github.com/getsentry/sentry/blob/3e0482c167a02a103b64f3313abf9a1cc58e8d15/src/sentry/incidents/logic.py#L665-L672), [update_alert_rule](https://github.com/getsentry/sentry/blob/3e0482c167a02a103b64f3313abf9a1cc58e8d15/src/sentry/incidents/logic.py#L945-L949), and the create/update serializers ([here](https://github.com/getsentry/sentry/blob/3e0482c167a02a103b64f3313abf9a1cc58e8d15/src/sentry/incidents/serializers/alert_rule.py#L515-L525),[ here](https://github.com/getsentry/sentry/blob/3e0482c167a02a103b64f3313abf9a1cc58e8d15/src/sentry/incidents/serializers/alert_rule.py#L547-L557))